### PR TITLE
Change Functions: rem_winter, RE_format, PS_format

### PR DESCRIPTION
**Pull request by rawann31**
## This pull request introduces enhancements to date handling and image filtering: 

1. **Date Conversion**: Simplified date handling by using `datetime.strptime` to convert image IDs to date objects.
2. **Winter Image Filtering**: Improved the filtering of winter images based on day-of-year calculations.
3. **Code Cleanup**: Removed redundant functions and improved code readability.

#### 1. Date Conversion
- **Updated Date Parsing**:
  - **RE_format**: Converts image ID with format `"%Y-%m-%dT%H%M%S"` to a date object.
  - **PS_format**: Converts image ID with format `"%Y%m%dT%H%M%S"` to a date object.

- **Day-of-Year Calculation**: Converts the date to the day number in the year using `time.timetuple().tm_yday`.

#### 2. Winter Image Filtering
- **Function**: `rem_winter`
  - **Purpose**: Filters out images from winter months (October 16 to May 15) based on their day-of-year.
  - **Days of Year**:
    - October 16 = 290th day
    - May 15 = 136th day

- **Logic**: Checks if the image date is not between these days and adds to the list of IDs to keep.

- **Removed Function**: The `list_to_string` function was deleted as it was deemed unnecessary.

#### 3. Code Example

```python
from datetime import datetime

# RE_format: Converts RE image ID to date object
# Example ID: 2018-12-03T083453_RE4_1B_band1.tif
def RE_format(image_id):
    time = image_id[:17] 
    time = datetime.strptime(time, "%Y-%m-%dT%H%M%S")
    return time

# PS_format: Converts PS image ID to date object
# Example ID: 20181203T083453_PS_1B_band1.tif
def PS_format(image_id):
    time = image_id[:15] 
    time = datetime.strptime(time, "%Y%m%dT%H%M%S")
    return time

# Function to remove images of winter months (Oct 16 to May 15)
# Oct 16 = 290 day of year , May 15 = 136 day of year
# Returns ID array of images after removing winter months
def rem_winter(ids = []):
   clear_ids = []
   
   for i in range(len(ids)):
       
        # Reformat current min for comparison
        # Check for RE image
        if ids[i][4] == '-':
            time = RE_format(ids[i])
            
        # Else is PS image
        else:
            time = PS_format(ids[i])
            
        day_of_year = time.timetuple().tm_yday # Get the day of the year from the datetime object

        if not(day_of_year >= 290) and not(day_of_year <= 136):
            clear_ids.append(ids[i])
   return clear_ids
```

**This pull request is part of Google Summer of Code 2024.**

### CODE REVIEW

File: DeeringAutoDownloadCode.py [hunk 1]
```diff
@@ -371,41 +374,12 @@ def rem_winter(ids = []):
...
-        else:
-            removal.append(0)
-            
-    # Create new array to return
-    clear_ids = []
-    
-    # Add each non-winter element to new array
-    for i in range(0, len(ids)):
-        if removal[i] == 0:
+        if not(day_of_year >= 290) and not(day_of_year <= 136):
```
Comment (fwitmer) [inline DeeringAutoDownloadCode.py]: I think this line would be clearer written without the negatives and with the smaller day first:
`        if day_of_year > 136 and day_of_year < 290:`
Do you agree?

**Review by fwitmer — CHANGES_REQUESTED** [review]
Please see the comment regarding the if statement on line 379 and let me know what you think.

File: DeeringAutoDownloadCode.py (hunk 1 above)
Comment (fwitmer) [reply DeeringAutoDownloadCode.py]: Or, we could use a chained comparison:
`     if 136 < day_of_year < 290:`

**Comment by rawann31** [comment]
Yes I totally agree with you in this

**Review by fwitmer — APPROVED** [review]
The revised chained comparison for day_of_year looks good.

**Pull request closed without merging**